core/app/models/concerns/spree/soft_deletable.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'discard'
+
+module Spree
+  module SoftDeletable
+    extend ActiveSupport::Concern
+
+    included do
+      include Discard::Model
+      self.discard_column = :deleted_at
+
+      default_scope { kept }
+    end
+
+    module ClassMethods
+      def with_deleted
+        Spree::Deprecation.warn <<-WARN.strip_heredoc, caller
```

Comment:
Not really significant, but I assume our Ruby version support is now such that we can use squiggly heredocs, making the `.strip_heredoc` unnecessary in this file.